test/helpers/add_retries_test_helper.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# Usage: Extend a descendant of Minitest::Test with this module
+# and then add "__Nretries" to the end of the names of the tests you want to retry up to N times if they fail
+module AddRetriesTestHelper
+  def run_one_method(klass, method_name, reporter)
+    match_data = method_name.match(/__(?<retries>\d+)retries$/)
+    return super unless match_data && match_data[:retries]
+    retries = match_data[:retries].to_i
+    result = nil
+
+    (retries + 1).times do |i|
+      result = report_retried_failures(retry_num: i, total_retries: retries, last_result: result) do
+        Minitest.run_one_method(self, method_name)
+      end
+      break if result.passed?
+    end
+
+    reporter.record(result)
+  end
+
+  # ENV["VERBOSE"] example:
+  #  KubernetesDeployTest#test_hpa_can_be_successful_and_gets_pruned__2retries 0.58 = F
+  #  KubernetesDeployTest#test_hpa_can_be_successful_and_gets_pruned__2retries (Retry 1/2) 0.53 = F
+  #  KubernetesDeployTest#test_hpa_can_be_successful_and_gets_pruned__2retries (Retry 2/2) 0.52 = F
```

Comment:
This is really annoying to achieve, as you can see below. In case a visual might help, here are the parts we're needing to print to look consistent with the final result printed by the actual verbose reporter:

![image](https://user-images.githubusercontent.com/4789493/53675351-6d053f00-3c63-11e9-968e-ef490d07a716.png)